src/ducks/sharing/index.js

```diff
@@ -0,0 +1,181 @@
+import Toggle from 'cozy-ui/react/Toggle'
+import classnames from 'classnames'
+import { Tab, Tabs, TabList, TabPanels, TabPanel } from 'cozy-ui/react/Tabs'
+// import Alerter from 'cozy-ui/react/Alerter'
+
+/* global cozy */
+export const filterSharedDocuments = (ids, doctype) =>
+  findPermSets(ids, doctype).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+
+export const findPermSet = (id, doctype) =>
+  findPermSets([id], doctype).then(sets => sets.length === 0 ? undefined : sets[0])
+
+// TODO: move this to cozy-client-js
+// there is cozy.client.files.getCollectionShareLink already, but I think that
+// there is a need of a bit of exploratory work and design for that API...
+
+export const findPermSets = (ids, doctype) =>
+  cozy.client.fetchJSON('GET', `/permissions/doctype/${doctype}/sharedByLink`)
+    .then(sets => sets.filter(set => {
+      const perm = set.attributes.permissions.collection
+      return perm.type === doctype && ids.find(id => perm.values.indexOf(id) !== -1) !== undefined
+    }))
+
+export const createPermSet = (id, doctype) =>
+  cozy.client.fetchJSON('POST', `/permissions?codes=email`, {
+    data: {
+      type: 'io.cozy.permissions',
+      attributes: {
+        permissions: {
+          files: {
+            type: 'io.cozy.files',
+            verbs: ['GET'],
+            values: [id],
+            selector: 'referenced_by'
+          },
+          collection: {
+            type: doctype,
+            verbs: ['GET'],
+            values: [id]
+          }
+        }
+      }
+    }
+  })
+
+export const deletePermSet = (setId) =>
+  cozy.client.fetchJSON('DELETE', `/permissions/${setId}`)
+
+export const getShareLink = (id, perms) =>
+  `${window.location.origin}/public?sharecode=${perms.attributes.codes.email}&id=${id}`
+
+export class ShareModal extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      loading: true,
+      creating: false,
+      permissions: null,
+      active: false,
+      copied: false
+    }
+  }
+
+  componentDidMount () {
+    const { id } = this.props.document
+    findPermSet(id, 'io.cozy.files')
+      .then(permissions => {
+        if (permissions !== undefined) {
+          this.setState({ loading: false, permissions, active: true })
+        } else {
+          this.setState({ loading: false })
+        }
+      })
+      .catch(() => this.onError())
+  }
+
+  toggleShareLink (active) {
+    const { id } = this.props.document
+    if (active) {
+      this.setState({ creating: true })
+      createPermSet(id, 'io.cozy.files')
+        .then(permissions => this.setState({ active, permissions, creating: false }))
+        .catch(() => this.onError())
+    } else {
+      const setId = this.state.permissions._id
+      this.setState({ active, permissions: null })
+      deletePermSet(setId)
+        .catch(() => this.onError())
+    }
+  }
+
+  onError () {
+    // Alerter.error('Error.generic')
```

Comment:
I don't know what to use here:
- `cozy-ui/react/Alerter`?
- `cozy-photos-v3/src/components/Alerter`?